types/seat/wlr_seat_keyboard.c

```diff
@@ -370,6 +370,13 @@ static void seat_client_send_keymap(struct wlr_seat_client *client,
 			continue;
 		}
 
+		if (keyboard->keymap == NULL) {
+			wl_keyboard_send_keymap(resource,
+				WL_KEYBOARD_KEYMAP_FORMAT_NO_KEYMAP, keymap_fd, 0);
```

Comment:
I'm assuming that there isn't a way for us to skip sending an fd here (-1?), and that we're allocating this fd just to send a legal but useless value across?

Reply:
Yes, libwayland errors out if we send -1.